src/main/kotlin/me/zeroeightsix/kami/feature/module/DiscordStatus.kt

```diff
@@ -0,0 +1,100 @@
+package me.zeroeightsix.kami.feature.module
+
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.KamiMod
+import me.zeroeightsix.kami.event.TickEvent
+import net.arikia.dev.drpc.DiscordEventHandlers
+import net.arikia.dev.drpc.DiscordRPC
+import net.arikia.dev.drpc.DiscordRichPresence
+import net.arikia.dev.drpc.DiscordUser
+import net.minecraft.util.Util.OperatingSystem
+import net.minecraft.util.Util.getOperatingSystem
+
+@Module.Info(name = "DiscordRPC", category = Module.Category.MISC, description = "Discord Rich presence")
+object DiscordStatus : Module() {
+    @Setting(name = "Version")
+    private var version = true
+
+    @Setting(name = "Username")
+    private var username = true
+
+    @Setting(name = "Server")
+    private var server = true
+
+    @Setting(name = "Flair")
+    private var message = true
+
+    private const val updateLimit: Long = 15000L // 15 seconds, should be good. (prevents api rate limiting)
+    private const val applicationId = "753664640789118999"
+
+    val messages = arrayOf(
+            "kami red how???",
+            "its really here",
+            "bigrat.monster!!!",
+            "free and open source ;)",
+            "kami kami kami kami kami",
+            "Scythe goddard mass gleaning",
+            "i have a scoop!",
+            "Not unfounded.",
+            getOperatingSystemMessage(),
+            "YES!!!!!!",
+            "086 is NOT A PRIME NUMBER!",
+            "3 is a PRIME NUMBER",
+            "Also try our sister mod KAMI BLUE!",
+            ":) (smiley face)",
+            "Read rat ode by elizabeth acevedo"
+    )
+
+    var lastUpdate: Long = 0L
+
+    override fun onEnable() {
+        initDiscord()
+    }
+
+    override fun onDisable() {
+        DiscordRPC.discordShutdown()
+    }
+
+    @EventHandler
+    private val updateListener =
+            Listener<TickEvent.Client.InGame>({
+                if (lastUpdate == 0L) lastUpdate = System.currentTimeMillis()
```

Comment:
If the user disables DiscordRPC for, let's say, 30 minutes and then re-enables it, this check won't actually work (as `lastUpdate` was set to the current time 30 minutes ago).

`lastUpdate` will be really behind then, and will actually cause this event listener to 'catch' up by spamming a RPC update to discord each of the following **120** ticks!! (30 mins = 1800 seconds = 120 x 15 seconds)